fraim/workflows/code/workflow.py

```diff
@@ -166,8 +166,9 @@ async def triage_with_semaphore(vuln: sarif.Result) -> Optional[sarif.Result]:
     async def workflow(self, input: CodeInput) -> List[sarif.Result]:
         """Main Code workflow - full control over execution with multi-step processing."""
         try:
-            # 1. Setup project input using utility
+            # 1. Setup project input using utility and initialize triager step
             self.project = self.setup_project_input(input)
+            self._triager_step = self.triager_step
```

Comment:
Not sure I follow the race condition.

You wrote:

>  but it gets generated inside the multiprocessing step, leaving it open to a race condition where multiple processes end up in the triager instantiation step,

Multiprocessing is separate processes, not threads --- i.e., no shared memory.  So the triager step does need to be initialized separately in each process.